static/js/controllers/reports.js

```diff
@@ -51,135 +51,88 @@ var _ = require('underscore'),
         }
       };
 
-      var setSelected = function(report) {
-        $scope.setSelected(report);
+      var setSelected = function(model) {
```

Comment:
It's a way of grouping stuff together. Say if we're showing a list of reports and some are expanded and some are not we can either store `expanded: true` on the db doc itself or create a new wrapper object (which I'm calling a model) and store `{ doc: doc, expanded: true }`. This has a couple of advantages - it keeps the db doc the same as what you see in the db, and it means we can use optimistic locking properly and not worry about accidentally persisting the `expanded: true` property. It also means we can have a model generator that produces an object which can be consumed by a generic template for use on multiple pages, so we don't have each controller doing its own thing.

Happy to talk about it in more depth.